### DISCUSSION TITLE
New serialization format

### DISCUSSION THREAD

**Discussion by IpsmLorem** [FiloSottile/age#359] [General]
Hi.
Did you consider using a more "modern" ascii serialization format like https://saltpack.org ? The format tends to be less prone to error when copy/pasting on Internet services.

**Comment by Tronic** [comment]
I would not necessarily call Saltpack modern, but I guess hence the quotes. Their encoding is difficult to calculate and takes more space than Base64, with dubious benefits (two slashes of base64 could be interpreted as italics on some messaging systems, if you forget to use code tags). Presumably you could avoid even that problem by picking some other character in place of slashes, while still using otherwise standard base64.

Not that I am suggesting to, but a modern solution might be using Unicode, a lot more of it than only 64, because it is becoming irrelevant how many bytes are consumed (assuming UTF-8 encoding) and only the number of characters matters. For instance, a base256 encoding where each byte of binary data turns into a character of Unicode. That could certainly avoid all formatting characters, path separators and such, too, while still restricting itself to glyphs available in common fonts (barring any legacy text consoles).
